ksql-engine/src/main/java/io/confluent/ksql/util/AvroUtil.java

```diff
@@ -188,37 +188,32 @@ private static AbstractStreamCreateStatement addAvroFields(
     return abstractStreamCreateStatement.copyWith(elements, properties);
   }
 
-
-  public static void validatePersistentQueryResults(
+  public static boolean isValidSchemaEvolution(
       final PersistentQueryMetadata persistentQueryMetadata,
       final SchemaRegistryClient schemaRegistryClient
   ) {
+    if (persistentQueryMetadata.getResultTopicSerde() != DataSource.DataSourceSerDe.AVRO) {
+      return true;
+    }
 
-    if (persistentQueryMetadata.getResultTopicSerde() == DataSource.DataSourceSerDe.AVRO) {
-      final org.apache.avro.Schema avroSchema = SchemaUtil.buildAvroSchema(
-          persistentQueryMetadata.getResultSchema(),
-          persistentQueryMetadata.getResultTopic().getName()
-      );
+    final org.apache.avro.Schema avroSchema = SchemaUtil.buildAvroSchema(
+        persistentQueryMetadata.getResultSchema(),
+        persistentQueryMetadata.getResultTopic().getName()
+    );
 
-      final String topicName = persistentQueryMetadata.getResultTopic().getTopicName();
+    final String topicName = persistentQueryMetadata.getResultTopic().getTopicName();
 
-      if (!isValidAvroSchemaForTopic(topicName, avroSchema, schemaRegistryClient)) {
-        throw new KsqlException(String.format(
-            "Cannot register avro schema for %s since it is not valid for schema registry.",
-            persistentQueryMetadata.getResultTopic().getKafkaTopicName()
-        ));
-      }
-    }
+    return isValidAvroSchemaForTopic(topicName, avroSchema, schemaRegistryClient);
   }
 
-
   private static boolean isValidAvroSchemaForTopic(
       final String topicName,
       final org.apache.avro.Schema avroSchema,
       final SchemaRegistryClient schemaRegistryClient
   ) {
     try {
-      return schemaRegistryClient.testCompatibility(topicName, avroSchema);
+      return schemaRegistryClient.testCompatibility(
```

Comment:
This looks to have been a bug in the old version where it checked compatibility with the wrong subject!!!